src/products/services/products/products.service.ts

```diff
@@ -0,0 +1,131 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { ProductDocument } from 'src/products/types/products.model'
+
+@Injectable()
+export class ProductsService {
+  constructor(
+    @InjectModel('Product') private productModel: Model<ProductDocument>,
+  ) {}
+
+  async create(
+    productId: string,
+    productType: [],
+    gender: [],
+    productShoeSize: [],
+    productSize: [],
+    productSizePantsWaist: [],
+    productSizePantsInseam: [],
+    productDescriptionOptional: string,
+    productImage: string,
+    ): Promise<any> {
+    const newProduct = new this.productModel({
+      productId,
+      productType,
+      gender,
+      productShoeSize,
+      productSize,
+      productSizePantsWaist,
+      productSizePantsInseam,
+      productDescriptionOptional,
+      productImage,
+    });
+    const result = await newProduct.save();
+    return result._id;
+  }
+
+  async findAll(): Promise<ProductDocument[]> {
+    const products = await this.productModel.find().exec();
+    if (!products || products.length === 0) {
+      throw new NotFoundException('No products found');
+    }
+    return products;
+  }
+
+  async findOne(id: string): Promise<ProductDocument> {
+    const product = await this.productModel.findById(id).exec();
+    if (!product || product === null) {
+      throw new NotFoundException(`Product ${id} not found`);
+    }
+    return {
+      productId: product.productId,
+      productType: product.productType,
+      gender: product.gender,
+      productShoeSize: product.productShoeSize,
+      productSize: product.productSize,
+      productSizePantsWaist: product.productSizePantsWaist,
+      productSizePantsInseam: product.productSizePantsInseam,
+      productDescriptionOptional: product.productDescriptionOptional,
+      productImage: product.productImage,
+    } as ProductDocument;
+  }
+
+  async updateProduct(
+    id: string, 
+    productId: string,
+    productType: [],
+    gender: [],
+    productShoeSize: [],
+    productSize: [],
+    productSizePantsWaist: [],
+    productSizePantsInseam: [],
+    productDescriptionOptional: string,
+    productImage: string,
+    ) {
+    const updatedProduct = await this.productModel.findById(id).exec();
+    if (!updatedProduct || updatedProduct === null) {
+      throw new NotFoundException(`Product ${id} not found`);
+    }
+    if (productId) {
+      console.log('productId ', productId);
```

Comment:
console.log can be refactored to use logging, this way we have a better control of how much information we want the system to show based on environments (production, staging, development) and situation (bug fix). You don't have to make this change in this pull request but let's create a ticket so we don't forget when the fall quarter starts. 
https://docs.nestjs.com/techniques/logger